src/osal/osal_freertos.h

```diff
@@ -62,12 +62,20 @@ TU_ATTR_ALWAYS_INLINE static inline void osal_task_delay(uint32_t msec)
 //--------------------------------------------------------------------+
 // Semaphore API
 //--------------------------------------------------------------------+
+#if configSUPPORT_STATIC_ALLOCATION == 1
 typedef StaticSemaphore_t osal_semaphore_def_t;
+#else
+typedef SemaphoreHandle_t osal_semaphore_def_t;
+#endif
 typedef SemaphoreHandle_t osal_semaphore_t;
 
 TU_ATTR_ALWAYS_INLINE static inline osal_semaphore_t osal_semaphore_create(osal_semaphore_def_t* semdef)
 {
+#if configSUPPORT_STATIC_ALLOCATION == 1
   return xSemaphoreCreateBinaryStatic(semdef);
+#else
+  return xSemaphoreCreateBinary();
```

Comment:
this will throw unused parameter warnings with semdef

Reply:
My bad sorry...  I have added (void) parameter...

New commit to follow...

Cheers,

Pete